### PR TITLE
Add .NET 6 exceptions

### PR DESCRIPTION
**Pull request by gewarren**
Contributes to dotnet/docs#21239.

Preview links:

- [CellBorderStyle](https://review.docs.microsoft.com/en-us/dotnet/api/system.windows.forms.tablelayoutpanel.cellborderstyle?view=net-5.0&branch=pr-en-us-5257#System_Windows_Forms_TableLayoutPanel_CellBorderStyle)
- [GrowStyle](https://review.docs.microsoft.com/en-us/dotnet/api/system.windows.forms.tablelayoutsettings.growstyle?view=net-5.0&branch=pr-en-us-5257)

### CODE REVIEW

File: xml/System.Windows.Forms/TableLayoutSettings.xml [hunk 1]
```diff
@@ -357,7 +357,8 @@
   
  ]]></format>
         </remarks>
-        <exception cref="T:System.ArgumentException">The property value is not valid for the enumeration type.</exception>
+        <exception cref="T:System.ArgumentException">.NET 5 and earlier versions only: The property value is not valid for the enumeration type.</exception>
```
Comment (gewarren) [inline xml/System.Windows.Forms/TableLayoutSettings.xml:360]: The [breaking change issue](https://github.com/dotnet/docs/issues/21239) said it was ArgumentOutOfRangeException. Should this be changed?

File: xml/System.Windows.Forms/TableLayoutPanel.xml [hunk 2]
```diff
@@ -247,6 +246,7 @@
   
  ]]></format>
         </remarks>
+        <exception cref="T:System.ComponentModel.InvalidEnumArgumentException">.NET 6 and later versions only: The property value is not valid for the enumeration type.</exception>
```
Comment (gewarren) [inline xml/System.Windows.Forms/TableLayoutPanel.xml:249]: I didn't find CellBorderStyle in TableLayoutSettings, so added the exception here.

Comment (RussKie) [reply xml/System.Windows.Forms/TableLayoutPanel.xml:249]: `TableLayoutSettings.CellBorderStyle` is internal.

File: xml/System.Windows.Forms/TableLayoutSettings.xml (hunk 1 above)
Comment (RussKie) [reply xml/System.Windows.Forms/TableLayoutSettings.xml:360]: It was `ArgumentException` in .NET Framework, but was changed in .NET Core 3.0 timeframe (https://github.com/dotnet/winforms/pull/665), which was before my time and thus went undocumented. And now we're changing it again :)

File: xml/System.Windows.Forms/TableLayoutSettings.xml [hunk 3]
```diff
@@ -357,7 +357,8 @@
...
+        <exception cref="T:System.ComponentModel.InvalidEnumArgumentException">.NET 6 and later versions only: The property value is not valid for the enumeration type.</exception>
```
Comment (gewarren) [inline xml/System.Windows.Forms/TableLayoutSettings.xml:361]:
```suggestion
        <exception cref="T:System.ComponentModel.ArgumentOutOfRangeException">.NET Core 3.0 - 3.1 only: The property value is not valid for the enumeration type.</exception>
        <exception cref="T:System.ComponentModel.InvalidEnumArgumentException">.NET 6 and later versions only: The property value is not valid for the enumeration type.</exception>
```

File: xml/System.Windows.Forms/TableLayoutSettings.xml (hunk 1 above)
Comment (gewarren) [inline xml/System.Windows.Forms/TableLayoutSettings.xml:360]:
```suggestion
        <exception cref="T:System.ArgumentException">.NET Framework only: The property value is not valid for the enumeration type.</exception>
```

File: xml/System.Windows.Forms/TableLayoutSettings.xml [hunk 4]
```diff
@@ -357,7 +357,9 @@
   
  ]]></format>
         </remarks>
-        <exception cref="T:System.ArgumentException">The property value is not valid for the enumeration type.</exception>
+        <exception cref="T:System.ArgumentException">.NET Framework only: The property value is not valid for the enumeration type.</exception>
+        <exception cref="T:System.ComponentModel.ArgumentOutOfRangeException">.NET Core 3.0 - 3.1 only: The property value is not valid for the enumeration type.</exception>
```
Comment (gewarren) [inline xml/System.Windows.Forms/TableLayoutSettings.xml:361]:
```suggestion
        <exception cref="T:System.ArgumentOutOfRangeException">.NET Core 3.0 - 3.1 only: The property value is not valid for the enumeration type.</exception>
```

File: xml/System.Windows.Forms/TableLayoutSettings.xml [hunk 5]
```diff
@@ -357,7 +357,9 @@
   
  ]]></format>
         </remarks>
-        <exception cref="T:System.ArgumentException">The property value is not valid for the enumeration type.</exception>
+        <exception cref="T:System.ArgumentException">.NET Framework only: The property value is not valid for the enumeration type.</exception>
+        <exception cref="T:System.ArgumentOutOfRangeException">.NET Core 3.0 - 3.1 only: The property value is not valid for the enumeration type.</exception>
```
Comment (RussKie) [inline xml/System.Windows.Forms/TableLayoutSettings.xml:361]:
```suggestion
        <exception cref="T:System.ArgumentOutOfRangeException">.NET Core 3.0 and 3.1, .NET 5.0 only: The property value is not valid for the enumeration type.</exception>
```

**Review by RussKie — APPROVED** [review]

**Pull request merged**